src/document.h

```diff
@@ -137,9 +137,20 @@ struct Document {
 
     uint Background() { return rootgrid ? rootgrid->cellcolor : 0xFFFFFF; }
 
-    void InitWith(Cell *r, wxString filename) {
+    void InitWith(Cell *r, wxString filename, Cell *ics) {
         rootgrid = r;
-        selected = Selection(r->grid, 0, 0, 1, 1);
+        if(ics) {
+            Grid* ipg = ics->parent->grid;
+            if(ipg) {
+                foreachcellingrid(c, ipg) {
+                    if(c == ics) {
+                        selected = Selection(ipg, x, y, 1, 1);

```

Comment:
it's going to be slightly odd this only works with 1x1 selections? maybe you can save the selection size in the file (be sure to check with the version on read)

Reply:
It is very little extra code. And I think it makes for confusing UX not doing it, since selections can be any size.

Reply:
Ok, I revised it and put the information into the file header. Is this ok?